dev/TabView/TabView.cpp

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "TabView.h"
+#include "TabViewAutomationPeer.h"
+#include "DoubleUtil.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+#include "SharedHelpers.h"
+
+static constexpr double c_tabMinimumWidth = 48.0;
+static constexpr double c_tabMaximumWidth = 200.0;
+
+static constexpr wstring_view c_tabViewItemMinWidthName{ L"TabViewItemMinWidth"sv };
+static constexpr wstring_view c_tabViewItemMaxWidthName{ L"TabViewItemMaxWidth"sv };
+
+// TODO: what is the right number and should this be customizable?
+static constexpr double c_scrollAmount = 50.0;
+
+TabView::TabView()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_TabView);
+
+    SetDefaultStyleKey(this);
+
+    Loaded({ this, &TabView::OnLoaded });
+    SelectionChanged({ this, &TabView::OnSelectionChanged });
+    SizeChanged({ this, &TabView::OnSizeChanged });
+}
+
+void TabView::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected{ *this };
+
+    m_tabContentPresenter.set(GetTemplateChildT<winrt::ContentPresenter>(L"TabContentPresenter", controlProtected));
+    m_scrollViewer.set(GetTemplateChildT<winrt::FxScrollViewer>(L"ScrollViewer", controlProtected));
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        m_scrollViewerLoadedRevoker = scrollViewer.Loaded(winrt::auto_revoke, { this, &TabView::OnScrollViewerLoaded });
+    }
+}
+
+void TabView::OnTabWidthModePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateTabWidths();
+}
+
+winrt::AutomationPeer TabView::OnCreateAutomationPeer()
+{
+    return winrt::make<TabViewAutomationPeer>(*this);
+}
+
+void TabView::OnLoaded(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    UpdateTabContent();
+}
+
+void TabView::OnScrollViewerLoaded(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        m_scrollDecreaseButton.set(SharedHelpers::FindInVisualTreeByName(scrollViewer, L"ScrollDecreaseButton").as<winrt::RepeatButton>());
+        if (auto decreaseButton = m_scrollDecreaseButton.get())
+        {
+            m_scrollDecreaseClickRevoker = decreaseButton.Click(winrt::auto_revoke, { this, &TabView::OnScrollDecreaseClick });
+        }
+
+        m_scrollIncreaseButton.set(SharedHelpers::FindInVisualTreeByName(scrollViewer, L"ScrollIncreaseButton").as<winrt::RepeatButton>());
+        if (auto increaseButton = m_scrollIncreaseButton.get())
```

Comment:
>auto [](start = 12, length = 4)

And here, any time you are creating a local from a tracker_ref

Reply:
ok, that works!